src/ORM/DataObject.php

```diff
@@ -4028,14 +4014,24 @@ public function summaryFields()
         $rawFields = $this->config()->get('summary_fields');
 
         // Merge associative / numeric keys
+        // Key could be a mixture of key types for example
+        // base class $summary_fields = ['Title'] - key is 0
+        // extension class $summary-fields = ['Title' => 'Custom Title'] - key is 'Title'
+        // Sort by numeric keys first so that any associative keys will override them later
+        uksort($rawFields, function ($a, $b) {
+            if (is_int($a) && !is_int($b)) {
+                return -1;
+            } elseif (!is_int($a) && is_int($b)) {
+                return 1;
+            }
+        });
```

Comment:
Why does this need to be added? In https://github.com/silverstripe/silverstripe-framework/issues/11050#issuecomment-2297821474 you implied the summary fields stuff from `DataExtension` isn't necessary (I assume that's where this came from)?

Reply:
Also, is there an updated unit test associated with this change?

Reply:
Turns out my initial assessment of the summary fields stuff in DataExtension was wrong.

Unit tested already via https://github.com/silverstripe/silverstripe-framework/blob/5/tests/php/ORM/DataObjectTest.php#L1433C21-L1433C50 - custom label is set via https://github.com/silverstripe/silverstripe-framework/blob/5/tests/php/ORM/DataObjectTest/Team_Extension.php

Reply:
lol so it was only working because there just happens to be a `DataExtension` applied at a global level somewhere? That's hilarious. Glad we're fixing this.